python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -455,23 +570,162 @@ async def liveView_loop(self, exposure_time):
+                    self.config.minBackground
+                    <= background_level
+                    <= self.config.maxBackground
+                ):
+                    self.log.debug("Taking exposure.")
+                    exposure = await self.take_image(
+                        shutter,
+                        images_in_sequence,
+                        image_index,
+                        exposure_time,
+                        timestamp,
+                        image_name,
+                    )
+                    self.log.debug("Establishing exposure background level.")
+                    background_level = await self.establish_exposure_background(
+                        exposure
+                    )
+                    self.log.debug(
+                        f"Background level is {background_level} and a value between "
+                        f"{self.config.minBackground} and {self.config.maxBackground} "
+                        f"is expected."
+                    )
+                    new_exposure_time = exposure_time
+                    if background_level > self.config.maxBackground:
+                        new_exposure_time = exposure_time / 2.0
+                        if new_exposure_time < min_exp_time:
+                            new_exposure_time = min_exp_time
+                    elif background_level < self.config.minBackground:
+                        new_exposure_time = exposure_time * 2.0
+                        if new_exposure_time > max_exp_time:
+                            new_exposure_time = max_exp_time
+
+                    if (
+                        not (
+                            self.config.minBackground
+                            <= background_level
+                            <= self.config.maxBackground
+                        )
+                        and new_exposure_time == exposure_time
+                    ):
+                        self.log.warn(
+                            "Cannot take an exposure with a valid background level. Ignoring."
+                        )
+                        exposure = None
+                        break
+
+                if exposure:
+                    exposure.save(os.path.join(self.directory, image_name + ".fits"))
+
+                await self.camera.endTakeImage()
+                self.evt_endTakeImage.put()
+
+                # Schedule the next image such that it starts
+                # AUTO_EXP_IMAGE_INTERVAL seconds after the start of
+                # the previous image cycle.
+                now = time.time()
+                time_diff = now - timestamp
+                sleep_time = self.config.autoExposureInterval
+                if time_diff < self.config.autoExposureInterval:
+                    sleep_time = self.config.autoExposureInterval - time_diff
+                await asyncio.sleep(sleep_time)
+
+        except Exception as e:
+            self.log.error("Error in auto exposure loop.")
```

Comment:
Instead of logging error and exception, log the exception only. You also don't need the `e` from the `except` above:

```
except Exception:
    self.log.exception("Error in auto exposure loop.")
    await self.stop_autoexposure()
    self.fault(...)
```

Reply:
Oops, I did know that but somehow did not do that. Thanks for spotting this. Done.